app/src/main/java/de/westnordost/streetcomplete/quests/address/AddHousenumberForm.kt

```diff
@@ -40,22 +41,32 @@ class AddHousenumberForm : AbstractQuestFormAnswerFragment<HousenumberAnswer>()
     private var addButton: View? = null
     private var subtractButton: View? = null
 
-    private var isHousename: Boolean = false
+    enum class InterfaceMode {
+        HOUSENUMBER, HOUSENAME, HOUSENUMBER_AND_HOUSENAME
+    }
+    private var interfaceMode: InterfaceMode = InterfaceMode.HOUSENUMBER
 
     private var houseNumberInputTextColors: ColorStateList? = null
 
     private val isShowingHouseNumberHint: Boolean get() = houseNumberInputTextColors != null
 
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
         val view = super.onCreateView(inflater, container, savedInstanceState)
-        isHousename = savedInstanceState?.getBoolean(IS_HOUSENAME) ?: false
-        setLayout(if(isHousename) R.layout.quest_housename else R.layout.quest_housenumber)
+        val prevMode = savedInstanceState?.getString(INTERFACE_MODE)?.let { InterfaceMode.valueOf(it) }
+        interfaceMode = prevMode ?: InterfaceMode.HOUSENUMBER
+        setLayout(
+            when(interfaceMode) {
+                InterfaceMode.HOUSENUMBER -> R.layout.quest_housenumber
+                InterfaceMode.HOUSENAME -> R.layout.quest_housename
+                InterfaceMode.HOUSENUMBER_AND_HOUSENAME -> R.layout.quest_housename_and_housenumber
+            }
+        )
```

Comment:
That could be separated into a function that is also called from `switchToHouseName` and `switchToHouseNameAndHouseNumber`, so the layouts don't have to be repeated there. But I guess it's fine either way.

Reply:
Or this could be done in `setLayout`, which then not accepts a layout ID as parameter, but an `InterfaceMode`.

Reply:
<details><summary>I had the idea to suggest something similar. I decided not to because I don't think it's really necessary from a cost:benefit perspective.</summary>

- This is isolated code, so a little messiness is not going to spread and become a worse problem later; there's no harm in cleaning up this piece later, before making another change.

- This is user interface code, and I've learned to accept that it's OK for UI code to be a little messy. Same reason I've embraced the ideas behind tailwind css for websites. A lot of what you write really isn't reusable, and half the time it's going to be thrown away and rewritten later, anyway.
    - Admittedly, this is more "back of the front end" (managing state and data), where it's more important to write clean code.
</details>

To be clear, I'm not opposed to it. I just don't consider it a blocker for merging.

Reply:
I created new `setLayoutBasedOnInterfaceMode` function as `setMode` appears in other quests and it should be consistent